### PR TITLE
feat:[SSCA-948]: Added Spec for Scorecard APIs

### PR DESCRIPTION
**Pull request by shash-harness**
## Describe your changes
Have added the spec changes for Scorecard API, This is needed to integrate the functionality of Scorecards in SBOMs

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55743)

### CODE REVIEW

**Comment by shash-harness** [comment]
trigger smartchecks

**Review by deepharness — DISMISSED** [review]

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -435,12 +435,64 @@ paths:
         - artifact
       parameters:
         - $ref: '#/components/parameters/AccountHeader'
+  '/v1/org/{org}/project/{project}/orchestration/{orchestrate-id}/sbom-scorecard':
```
Comment (reetika04) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:438]: please make it orchestration-id to keep it consistent across? You can refer to other api's in SSCA

**Comment by shash-harness** [comment]
trigger uts

File: ssca-manager/contracts/openapi/v1/openapi.yaml (hunk 1 above)
Comment (shash-harness) [reply ssca-manager/contracts/openapi/v1/openapi.yaml]: Made the changes .

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -1191,20 +1243,169 @@ components:
...
+          x-stoplight:
+            id: 9ckvau79tna59
+        sbom:
+          $ref: '#/components/schemas/SbomDetailsScorecard'
+        score_card_info:
+          $ref: '#/components/schemas/ScorecardInfo'
+        scores:
+          type: array
+          items:
+            $ref: '#/components/schemas/Scores'
```
Comment (dhruv-harness) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:1408]: If the `scores` field is an array, the individual element should not be called `Scores`(plural)

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -1191,20 +1243,169 @@ components:
...
+        avg_score:
+          type: string
+          x-stoplight:
+            id: walgcdlqybmje
+        creation_on:
+          type: string
+          x-stoplight:
+            id: 9ckvau79tna59
+        sbom:
+          $ref: '#/components/schemas/SbomDetailsScorecard'
```
Comment (dhruv-harness) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:1402]: I think sbom is confusing, does it refer to the sbom or the sbomdetailsscorecard?

Comment (shash-harness) [reply ssca-manager/contracts/openapi/v1/openapi.yaml:1402]: The details are for SBOM for which this scorecard is generated, but there already exists a SbomDetails Model, so had added in this manner.

File: ssca-manager/contracts/openapi/v1/openapi.yaml (hunk 2 above)
Comment (shash-harness) [reply ssca-manager/contracts/openapi/v1/openapi.yaml:1408]: Will update this.

File: ssca-manager/contracts/openapi/v1/openapi.yaml (hunk 3 above)
Comment (dhruv-harness) [reply ssca-manager/contracts/openapi/v1/openapi.yaml:1402]: Then we can refactor the field name to match the dto name

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Review by dhruv-harness — APPROVED** [review]

**Comment by shash-harness** [comment]
trigger ut9

**Comment by shash-harness** [comment]
trigger ut6

**Comment by shash-harness** [comment]
trigger ut8

**Pull request merged**